lib/sshkit.ex

```diff
@@ -262,17 +263,97 @@ defmodule SSHKit do
     Enum.map(context.hosts, run)
   end
 
-  # def upload(context, path, options \\ []) do
-  #   …
-  #   # resolve remote relative to context path
-  #   remote = Path.expand(Map.get(options, :as, Path.basename(path)), _)
-  #   SCP.upload(conn, path, remote, options)
-  # end
-
-  # def download(context, path, options \\ []) do
-  #   …
-  #   remote = _ # resolve remote relative to context path
-  #   local = Map.get(options, :as, Path.basename(path))
-  #   SCP.download(conn, remote, local, options)
-  # end
+  @doc ~S"""
+  Upload a file or files to the given context.
+  Returns a list of `:ok`.
+  There is one `:ok` per connected host where file upload was successful.
+
+  ## Examples
+
+  Upload all files and folders in current directory to "/workspace".
+
+  ```
+  [:ok] =
+    "my.remote-host.tld"
+    |> SSHKit.context
+    |> SSHKit.path("/workspace")
+    |> SSHKit.upload(".", recursive: true)
+
+  ```
+
+  Upload file to different name on host
+
+  ```
+  [:ok] =
+    "my.remote-host.tld"
+    |> SSHKit.context
+    |> SSHKit.upload("some_file.txt", as: "other_file.txt" )
+
+  ```
+
+  """
+  def upload(context, path, options \\ []) do
+    # resolve remote relative to context path
+    # override with :as option if present
+    remote = case Keyword.get(options, :as, false) do
+      false -> case context.path do
+        nil -> path
+        _ -> Path.join(context.path, path)
+      end
```

Comment:
How about

```elixir
nil -> case context.path do
  nil -> path
  base -> Path.join(base, path)
end
```

because `base` will be bound to the value of `context.path`?

Maybe this is even something Credo should check? (Potentially open an issue there to detect when you're accessing the original value where you could be using the match?)